elastic/apm-server
# [java attacher] set each space-separated arg individually in cmd

## Motivation/summary

the *Command needs each argument to be space
separated in order to work. The previous code was
grouping flags+arguments as logical units, which
does not get correctly executed by the binary.

the config was also defined in an inconsistent way. Originally it was:

```yml
apm-server:
  java_attacher:
    enabled: true
    download-agent-version: 1.27.0
    discovery-rules:
    - include-main: opbeans
    - exclude-main: ".*"
    config:
      server_url: "http://localhost:8200"

```
It made sense, and simplified the code, to have `config` values be treated the same as `discover-rules`:
```yml
apm-server:
  java_attacher:
    enabled: true
    download-agent-version: 1.27.0
    discovery-rules:
    - include-main: opbeans
    - exclude-main: ".*"
    config:
    - server_url: "http://localhost:8200"
```
(config was a map, now it's a slice of maps)

## Checklist


## How to test these changes

0. Download the java-attacher jar if this isn't being tested with a release that already has it pre-packaged:
  ```
make build/apm-agent-attach-cli-1.27.0-slim.jar
cp build/java-attacher.jar .
```
1. Start apm-server with the above config, and `seccomp.enabled=false` and `log.level=debug`
2. Verify that there's output from the java-attacher
